### ISSUE TITLE
MSVC link error

### ISSUE THREAD

**Issue by Molochnik** [tesseract-ocr/tesseract#1144]
I have downloaded that latest 4.0 release, new leptonica and tried to buid it under MSVC 2017, unicode project. After several hours of I almost succeded in building the project but one problem is still remaining unresolved, I don't know how to fix it.

network.cpp (line 209):
`network = new Convolve(stub.name_, stub.ni_, 0, 0);`

MSVC gives the linker error:
 error LNK2001: unresolved external symbol "public: __thiscall tesseract::Convolve::Convolve(class STRING const &,int,int,int)" (??0Convolve@tesseract@@QAE@ABVSTRING@@HHH@Z)

If this line is commented, project is built normally without errors.
convolve,cpp is included and Convolve::Convolve is definetely presented.
Weird thing that there is another file - networkbuilder.cpp where
one can see almost the same code, and it works fine:
```
  Convolve* convolve =
      new Convolve("Convolve", input_shape.depth(), x / 2, y / 2);

```
But after being copied to network.cpp it give the same link error.
I spent an hour trying to fix but failed.
What if I just comment this line? When it is used?

**Comment by amitdo** [comment]
Commenting that line is not a good idea.

https://msdn.microsoft.com/en-us/library/f6xx1b1z.aspx

CC: @egorpugin

**Comment by egorpugin** [comment]
Are you trying to build with cppan?

**Comment by Molochnik** [comment]
No, never heard of it. MSVC project, mostly assembled manually. The previos versions 3.02, 3.04, were assembled and compiled normally. 
First I thought that maybe it's leptonica's "convolve.c" that have collisions with lstm's "convolve.cpp" module and renamed the "convolve.c" to "convolve2.c" with no effect.

**Comment by egorpugin** [comment]
Please, see this page for build instructions https://github.com/tesseract-ocr/tesseract/wiki/Compiling#windows

**Comment by Molochnik** [comment]
cppan --build pvt.cppan.demo.google.tesseract.tesseract-master.
gives an error
boost::filesystem::path codecvt to wstring: error

**Comment by egorpugin** [comment]
What is you system version? Are you using windows 7?

**Comment by Molochnik** [comment]
WIndows 10, MSVC 2017 community

**Comment by egorpugin** [comment]
And what is your account name? Is it in russian?

**Comment by Molochnik** [comment]
Account to where? I have no account in Russian letters, though I am Russian

**Comment by egorpugin** [comment]
Windows user account.

**Comment by Molochnik** [comment]
Oh I see, I changed it to another user, now cppan goes further but no enough "There errors during test start". Log files are empty

**Comment by egorpugin** [comment]
Do you use the latest cmake?

**Comment by Molochnik** [comment]
I use the latest MSVC, is this enough for the latest cmake?
I solved the problem just by copied the source code from convolve.cpp (it's very small) to network.cpp and everything went fine.

**Comment by egorpugin** [comment]
It's ok to have manual fixes. But as you can see, windows build bots are fine. So there are some unusual things in your system setup.

**Comment by Molochnik** [comment]
When I was building 3.02 and 3.04 I didn't know about cppan, so i just manually added all required libraries and everything went fine. Today I found that there is 3.05 version, I downloaded it, added libs and evrything was compiled and linked perfectly as always, it took me around 15 minutes. Obviously I like that way and prefer to use it again with 4.0 version. 
As per my PC, I didn' do to it something unusual, OS  was installed a month ago, MSVC 2017, the tesseract code was not changed, As I said, I resolved the problem, so no need to do anything to the code, perhaps you are right and the problem is with my machine

**Comment by egorpugin** [comment]
> so i just manually added all required libraries 

It was added during 3.05.

> so i just manually added all required libraries 

Cppan is developed to do this for you.

**Comment by Molochnik** [comment]
Thank you for your efforts!
Ok, Egor, I will try to use cppan on a new virtual machine with Win10, pure MSVC and a proper Administrator.

**Comment by Molochnik** [comment]
After creating MSVC project again more accurately ()adding files one by oneI succeeded in  building MSVC project without changing the source code. Nevertheless one weird thing I made - I had to add convolve.cpp in the main project. Now it contains two files - main.cpp (my own source code) and convolve.cpp (other dependencies are through lib files). 
MSVC also gives a warning  while linking:
warning LNK4042: object specified more than once; extras ignored
I don't know what it means
